src/common/Debugging.h

```diff
@@ -0,0 +1,70 @@
+/*
+* Copyright (c) 2018 RISC Software GmbH
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#pragma once
+
+#include <string>
+#include <map>
+
+class TopoDS_Shape;
+
+namespace tigl
+{
+void dumpShape(const TopoDS_Shape& shape, const std::string& outputDir, const std::string& filename, int counter = -1);
+
+// represents a point in the execution of a program at which shapes can be dumped
+class TracePoint
+{
+public:
+    TracePoint(const std::string& outputDir);
+
+    void operator++(int);
+    void dumpShape(const TopoDS_Shape& shape, const std::string& filename);
+
+private:
+    std::string m_outputDir;
+    int m_counter;
+};
+
+// creates a trace point at the location of this macro with the specified variable name and output directory
+// the trace point is static and incremented each time programm flow executes over the location of this macro
+#define TRACE_POINT_OUTPUT_DIR(variableName, outputDir) static TracePoint variableName(outputDir); variableName++
```

Comment:
Maybe the increment operator is not the right "name" for the operation performed. What I want is the trace point to know, how many times execution has passed by. Therefore the trace point has static storage duration and is incremented each time execution passes. I need this information, because the files dumped subsequently need to be prefixed by a counter to not overwrite themselves each time they are dumped.